agent/handlers/gql/schema.go

```diff
@@ -32,11 +46,175 @@ func CreateSchema(
+				Resolve: logDriverResolver,
+			},
+			"LogOptions": &graphql.Field{
+				Type:    JSON,
+				Resolve: logOptionsResolver,
+			},
+			"ContainerARN": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: containerARNResolver,
+			},
+		},
+	})
+
+	var taskType = graphql.NewObject(graphql.ObjectConfig{
+		Name: "Task",
+		Fields: graphql.Fields{
+			"Cluster": &graphql.Field{
+				Type: graphql.String,
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					return cluster, nil
+				},
+			},
+			"TaskARN": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: taskARNResolver,
+			},
+			"Family": &graphql.Field{
+				Type: graphql.String,
+			},
+			"Revision": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: taskRevisionResolver,
+			},
+			"DesiredStatus": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: taskDesiredStatusResolver,
+			},
+			"KnownStatus": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: taskKnownStatusResolver,
+			},
+			"Limits": &graphql.Field{
+				Type:    JSON,
+				Resolve: taskLimitsResolver,
+			},
+			"PullStartedAt": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: taskPullStartedResolver,
+			},
+			"PullStoppedAt": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: taskPullStoppedResolver,
+			},
+			"ExecutionStoppedAt": &graphql.Field{
+				Type:    graphql.String,
+				Resolve: taskExecutionStoppedResolver,
+			},
+			"AvailabilityZone": &graphql.Field{
+				Type: graphql.String,
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					return availabilityZone, nil
+				},
+			},
+			"LaunchType": &graphql.Field{
+				Type: graphql.String,
+			},
+			"Containers": &graphql.Field{
+				Type: graphql.NewList(containerType),
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					if task, ok := p.Source.(*apitask.Task); ok {
+						containerNameToDockerContainer, ok := state.ContainerMapByArn(task.Arn)
+						if !ok {
+							return "", errors.Errorf("GraphQL task response: unable to get container name mapping for task %v",
+								task.Arn)
+						}
+						resp := []*apicontainer.DockerContainer{}
+						for _, dockerContainer := range containerNameToDockerContainer {
+							resp = append(resp, dockerContainer)
+						}
+						return resp, nil
+					}
+					return nil, nil
+				},
+			},
+		},
+	})
+
+	// Create GraphQL Schema
 	fields := graphql.Fields{
 		"Container": &graphql.Field{
-			Type:    graphql.String,
-			Resolve: dockerIDResolver,
+			Type: containerType,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				dockerContainer, ok := p.Context.Value(Container).(*apicontainer.DockerContainer)
+				if !ok {
+					return nil, errors.Errorf("GraphQL Schema Error: Could not cast to container")
+				}
+
+				return dockerContainer, nil
+			},
+		},
+		"Task": &graphql.Field{
+			Type: taskType,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
```

Comment:
Anythin thats doing extended logic like this makes more sense to be tucked as a named function